src/descriptor/tr.rs

```diff
@@ -529,23 +529,34 @@ fn parse_tr_tree(s: &str) -> Result<expression::Tree, Error> {
     }
 }
 
-fn split_once(inp: &str, delim: char) -> Option<(&str, &str)> {
+fn split_key_and_tree(inp: &str) -> Option<(&str, &str)> {
     if inp.is_empty() {
         None
     } else {
-        let mut found = inp.len();
-        for (idx, ch) in inp.chars().enumerate() {
-            if ch == delim {
-                found = idx;
-                break;
+        // hit the first comma when the open_bracket == 0, we can split at that point
+        let mut open_brackets = 0;
+        let mut ind = 0;
+        for c in inp.chars() {
+            if c == '(' {
+                open_brackets += 1;
+            } else if c == ')' {
+                open_brackets -= 1;
+            } else if c == ',' {
+                if open_brackets == 0 {
+                    break;
+                }
             }
+            ind += 1;
```

Comment:
Not thrilled with this part. Ideally, we should be able to get the complete Tree and parse the script spend and key spend nicely. This hack worked nicely for key spend when there was a single key and now it's an even ugly hack. I will fixup the tree decoding later